src/read.c

```diff
@@ -4670,25 +4688,29 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
                         // One of the following invalid cases:
                         // * Multiple items are claiming to be the alpha auxiliary of the current item.
                         // * Alpha auxiliary is dimg for another item.
-                        avifFree(alphaItemIndices);
+                        avifFree(dimgIdxToAlphaItemIdx);
                         *alphaItem = NULL;
                         *isAlphaItemInInput = AVIF_FALSE;
                         return AVIF_RESULT_INVALID_IMAGE_GRID;
                     }
-                    alphaItemIndices[alphaItemCount++] = j;
+                    if (item->dimgIdx >= tileCount || dimgIdxToAlphaItemIdx[item->dimgIdx] != itemIndexNotSet) {
```

Comment:
It seems that the `dimgIdxToAlphaItemIdx[item->dimgIdx] != itemIndexNotSet` check is redundant with the `seenAlphaForCurrentItem` check and can be omitted. Please double check.

Reply:
It is not the same check: `dimg` vs `auxl`